src/backend/utils/fetch_with_cookie.ts

```diff
@@ -0,0 +1,18 @@
+import { DEV_PORTAL_URL } from 'common/constants'
+import getPartitionCookies from './get_partition_cookies'
+
+export async function fetchWithCookie(url: string, method: string) {
+  const cookieString = await getPartitionCookies({
+    partition: 'persist:auth',
+    url: DEV_PORTAL_URL
+  })
```

Comment:
we should probably return the fetch to handle `result.ok` in the implementations

```ts

const response = await fetchWithCookie(url, 'POST')

if (!response.ok) {
  ...
}

```

Reply:
hmm this could get pretty repetitive though. what do you think about adding this in the fetchWithCookie function